### PR TITLE
Add jerry-client.py

### PR DESCRIPTION
**Pull request by polaroi8d**
Add the client side of the communication to debugger. 

JerryScript-DCO-1.0-Signed-off-by: Levente Orban orbanl@inf.u-szeged.hu

### CODE REVIEW

File: jerry-debugger/jerry-client.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+def main():
+    try:
+        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client_socket.connect((HOST, PORT))
+    except socket.error, msg:
+        print('Failed to create the socket. Error code: %d, Error message: %s' % (msg[0], msg[1]))
+        sys.exit()
+
+    print ('Socket created on: %d'% PORT)
```
Comment (zherczeg) [inline jerry-debugger/jerry-client.py]: What is the style: space or not space after print?

Comment (polaroi8d) [reply jerry-debugger/jerry-client.py]: Without space, thanks.

File: jerry-debugger/jerry-client.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+PORT = 5001
+HOST = "localhost"
+
+def main():
+    try:
+        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client_socket.connect((HOST, PORT))
+    except socket.error, msg:
+        print('Failed to create the socket. Error code: %d, Error message: %s' % (msg[0], msg[1]))
+        sys.exit()
```
Comment (stampho) [inline jerry-debugger/jerry-client.py]: Return code is missing from here. Probably use sys.exit(1) or you might use the error code above but not sure whether is this the common way to do this.

Hint: you can easily check the sys.exit() usage in the Jerry's python scripts with the following command:
find . -name "*.py" | xargs grep --color "sys.exit"

File: jerry-debugger/jerry-client.py (hunk 1 above)
Comment (stampho) [inline jerry-debugger/jerry-client.py]: nitpick: style issues with space:
1. print (
2. %d'%

**Review by stampho — CHANGES_REQUESTED** [review]

File: jerry-debugger/jerry-client.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+PORT = 5001
+HOST = "localhost"
+
+def main():
+    try:
+        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client_socket.connect((HOST, PORT))
+    except socket.error, msg:
+        print('Failed to create the socket. Error code: %d, Error message: %s' % (msg[0], msg[1]))
```
Comment (stampho) [inline jerry-debugger/jerry-client.py]: one more thing: probably you should send error message to the stderr. Check sys.stderr.write in the python documentation.

File: jerry-debugger/jerry-client.py (hunk 2 above)
Comment (polaroi8d) [reply jerry-debugger/jerry-client.py]: Thanks for the instruction.

File: jerry-debugger/jerry-client.py (hunk 1 above)
Comment (polaroi8d) [reply jerry-debugger/jerry-client.py]: :+1:

File: jerry-debugger/jerry-client.py (hunk 3 above)
Comment (polaroi8d) [reply jerry-debugger/jerry-client.py]: Okey.

File: jerry-debugger/jerry-client.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+PORT = 5001
+HOST = "localhost"
+
+def main():
+    try:
+        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client_socket.connect((HOST, PORT))
+    except socket.error, msg:
+        print >> sys.stderr, ('Failed to create the socket. Error code: %d, Error message: %s' % (msg[0], msg[1]))
```
Comment (stampho) [inline jerry-debugger/jerry-client.py]: Most probably this won't work with python3. I would propose to use sys.stderr.write('message\n')

Anyway, I've just found something: http://stackoverflow.com/questions/2949974/how-to-exit-a-program-sys-stderr-write-or-print

It seems sys.exit('message') is a proposed way and exits with exit code 1

**Comment by stampho** [comment]
LGTM

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**